sdk/core/azure-core/src/http/request.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <string>
+
+#include <http/http.hpp>
+
+using namespace azure::core::http;
+using namespace std;
+
+std::string Header::getName() { return this->name; }
+std::string Header::getValue() { return this->value; }
+
+http_method::HttpMethod Request::getMethod() { return this->method; }
+string Request::getUrl() { return this->url; }
+string Request::getBody() { return this->body; }
+vector<Header> Request::getHeaders() { return this->headers; }
+
+void Request::addHeader(string name, string value)
+{
+  try
+  {
+    this->headers.push_back(Header(name, value));
+  }
+  catch (exception e)
+  {
+    throw e;
+  }
```

Comment:
What is the point of catching the exception and throw it?

Reply:
I was thinking that in the future we might want to log the exception of do something else.
But I think I can remove it for now and wait.

Reply:
Okay, but not instead of `throw e;`, you may want `throw;`, which is to rethrow the original exception.